File: server/Content/FlashDeck.js
class FlashDeck {
  constructor(deckname, topic, cards = []) {
    this.deckname = deckname;
    this.topic = topic;
    this.listOfCards = cards; // list of Flashcard
    this.index = 0;
    this.maxSize = 100;
  }

  addCard(card) {
    if (this.listOfCards.length < this.maxSize) {
      this.listOfCards.push(card);
      return true;
    }
    return false;
  }

  deleteCard(card) {
    this.listOfCards = this.listOfCards.filter(c => c !== card);
  }

  hasNext() {
    return this.index < this.listOfCards.length;
  }

  next() {
    if (this.hasNext()) {
      return this.listOfCards[this.index++];
    }
  }

  reset() {
    this.index = 0;
  }

  iterate() {
    while (this.hasNext()) {
      let currentCard = this.next();
      currentCard.flip();
    }
  }

  scramble() {
    this.listOfCards.sort(() => Math.random() - 0.5);
    this.index = 0;
  }
}

File: server/Content/FlashDeck.py
import Flashcard

class FlashDeck:

    def __init__(self, deckname, topic, cards):
        self.deckname = deckname
        self.topic = topic
        self.listOfCards = cards #list of Flashcard
        self.index = 0
        self.maxSize = 100

    def addCard(self, card):
        if len(self.listOfCards) < self.maxSize:
            self.listOfCards.append(card)
            return True
        else:
            return False

    def deleteCard(self, card):
        pass

    def hasNext(self):
        if(self.index < len(self.listOfCards)):
            return True
        else:
            return False
    
    def next(self):
        if(self.hasNext()):
            self.index = self.index + 1
            return self.listOfCards[self.index -1]
        
    def reset(self):
        self.index = 0

    def iterate(self):
        while(self.hasNext()):
            #delays?
            currentCard = self.next()
            currentCard.flip()

    def scramble(self):
        #scramble list, set index to 0
        pass